### PR TITLE
Port configmap test to node e2e suite

### LINKED ISSUE #26833: Port configmap tests to node e2e
The [configmap tests](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/configmap.go) should run as part of the node e2e suite. For now the tests can be duplicated in the `test/e2e_node` directory.

Reserved for @mnshaw

### PR DESCRIPTION
**Pull request by mnshaw**
fixes #26833

### CODE REVIEW

File: test/e2e_node/configmap.go [hunk 1]
```diff
@@ -0,0 +1,370 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
```
Comment (yujuhong) [inline test/e2e_node/configmap.go]: s/2014/2016

**Comment by yujuhong** [comment]
Test failed on `ubuntu-trusty-docker8-image`
```
â€¢ Failure [2.059 seconds]
[k8s.io] ConfigMap
/var/lib/jenkins/workspace/node-pull-build-e2e-test@3/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:660
  should be consumable from pods in volume as non-root with FSGroup [Feature:FSGroup] [It]
  /var/lib/jenkins/workspace/node-pull-build-e2e-test@3/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e_node/configmap.go:45

  Expected error:
      <*errors.errorString | 0xc8208e3050>: {
          s: "pod 'pod-configmaps-a2b3fda2-3195-11e6-bf48-42010af0004d' terminated with failure: &{ExitCode:-1 Signal:0 Reason:ContainerCannotRun Message: StartedAt:{Time:2016-06-13 18:35:43 +0000 UTC} FinishedAt:{Time:2016-06-13 18:35:43 +0000 UTC} ContainerID:docker://5f97183dfe0c2ea8c30da9667cbbf6a6f55906f7c6d101d6daca53170c29c9d2}",
      }
      pod 'pod-configmaps-a2b3fda2-3195-11e6-bf48-42010af0004d' terminated with failure: &{ExitCode:-1 Signal:0 Reason:ContainerCannotRun Message: StartedAt:{Time:2016-06-13 18:35:43 +0000 UTC} FinishedAt:{Time:2016-06-13 18:35:43 +0000 UTC} ContainerID:docker://5f97183dfe0c2ea8c30da9667cbbf6a6f55906f7c6d101d6daca53170c29c9d2}
  not to have occurred

  /var/lib/jenkins/workspace/node-pull-build-e2e-test@3/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:1993
```
I'm going to re-trigger the test to see if it's a flake

**Comment by yujuhong** [comment]
@k8s-bot node e2e test this: issue #IGNORE

**Comment by yujuhong** [comment]
@mnshaw, you'll have to sign the CLA by following the instructions in https://github.com/kubernetes/kubernetes/pull/27291#issuecomment-225665885 :-)

**Comment by yujuhong** [comment]
@mnshaw, this seems like a legit failure. Could you take a look at the logs and see what went wrong? Thanks!
https://console.cloud.google.com/storage/kubernetes-jenkins/pr-logs/pull/27291/node-pull-build-e2e-test/10352/

**Comment by mnshaw** [comment]
I signed it!

**Comment by yujuhong** [comment]
As for the failure in the unit/integration test (unrelated to to the jenkins gce node e2e failure):
```
Found illegal 'flag' usage. If these are false positives you should run `hack/verify-flags-underscore.py -e > hack/verify-flags/exceptions.txt` to update the list.
test/e2e_node/configmap.go:						Command: []string{"/mt", "--break_on_expected_content=false", "--retry_time=120", "--file_content_in_loop=/etc/configmap-volume/data-1"},
FAILED   ./hack/../hack/verify-flags-underscore.py	64s
```

Run `hack/verify-flags-underscore.py -e > hack/verify-flags/exceptions.txt` and amend your commit (or add a new commit) to include the change to `exceptions.txt` should solve the problem.

**Comment by yujuhong** [comment]
Based on @mnshaw's investigation, the test failed on the coreos image when FSGroup is set.

```
Jun 13 19:20:35 localhost kubelet[1254]: E0613 19:20:03.629104    1254 manager.go:2109] container start failed: RunContainerError: runContainer: Error response from daemon: Cannot start container a0da89d47c4914972d1176a9e01250cff5d82efcfc2fca58e3a18e3e0d971720: [8] System error: Failed to open group file: open /etc/group: no such file or directory
Jun 13 19:20:35 localhost kubelet[1254]: E0613 19:20:03.629237    1254 pod_workers.go:183] Error syncing pod d443416f-319b-11e6-95f2-42010af00008, skipping: failed to "StartContainer" for "configmap-volume-test" with RunContainerError: "runContainer: Error response from daemon: Cannot start container a0da89d47c4914972d1176a9e01250cff5d82efcfc2fca58e3a18e3e0d971720: [8] System error: Failed to open group file: open /etc/group: no such file or directory"
```

/cc @yifan-gu @euank, any idea?

**Comment by euank** [comment]
The simplest reproduction of that error is: `docker run --group-add=1001 gcr.io/google_containers/mounttest:0.6` which fails on the old CoreOS version used in the node-e2e (docker 1.8).

That same command runs fine on any of the current versions of CoreOS (docker 1.10). Presumably, docker changes its behavior with regards to images that have no `/etc/group` file in their rootfs. As a note, the above `docker run` works with images that have an `/etc/group` file even on older docker versions, so one possible workaround would be to modify the `mounttest` container to have `/etc/group` present.

Since we need to update the CoreOS image anyways, it seems sane to just depend on #27383 for this one and wait for the image to be updated.

**Comment by yujuhong** [comment]
@euank thanks for investigating the issue! Waiting for #27383 to land first SGTM.

**Comment by yujuhong** [comment]
@k8s-bot test this: issue #IGNORE

File: test/e2e_node/configmap.go [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+						},
+					},
+				},
+				RestartPolicy: api.RestartPolicyNever,
+			},
+		}
+
+		assignPodToNode(pod)
+
+		defer func() {
```
Comment (yujuhong) [inline test/e2e_node/configmap.go]: Don't need to delete the pod manually. The framework will delete the entire namespace after the test completes.

File: test/e2e_node/configmap.go [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+				Namespace: f.Namespace.Name,
+				Name:      name,
+			},
+			Data: map[string]string{
+				"data-1": "value-1",
+			},
+		}
+
+		By(fmt.Sprintf("Creating configMap with name %s", configMap.Name))
+		defer func() {
```
Comment (yujuhong) [inline test/e2e_node/configmap.go]: Don't need to delete the configmap manually. The framework will delete the entire namespace after the test completes.

**Comment by yujuhong** [comment]
@mnshaw, the blocker #27383 has been closed. Please address the comments.

File: test/e2e_node/configmap.go (hunk 1 above)
Comment (yujuhong) [inline test/e2e_node/configmap.go]: Ping. just a nit.

**Comment by yujuhong** [comment]
LGTM with a small nit from the previous review. @mnshaw, could you address the nit and squash the commits? Thanks!

**Comment by yujuhong** [comment]
The jenkins verification complained:
```
Verifying ./hack/../hack/verify-gofmt.sh
!!! 'gofmt -s -w' needs to be run on the following files: 
./test/e2e_node/configmap.go
FAILED   ./hack/../hack/verify-gofmt.sh	9s
```

@mnshaw, gofmt is not happy :)

**Comment by yujuhong** [comment]
LGTM. Thanks a lot for the PR!

**Comment by yujuhong** [comment]
@mnshaw congrats on the first PR!

/cc @timstclair

**Comment by mnshaw** [comment]
@yujuhong Thanks for all the help!

**Comment by timstclair** [comment]
Nice!

**Pull request merged**